ddtrace/internal/processor/trace.py

```diff
@@ -278,3 +279,33 @@ def on_span_finish(self, span):
                     if config._trace_compute_stats:
                         span.set_metric(SAMPLING_PRIORITY_KEY, USER_KEEP)
                     break
+
+
+class PeerServiceProcessor(SpanProcessor):
+    def __init__(self, peer_service_config):
+        self._config = peer_service_config
+        self.enabled = self._config.enabled
+
+    def on_span_start(self, span):
+        """
+        We don't do anything on span start
+        """
+        pass
+
+    def on_span_finish(self, span):
+        if not self.enabled:
+            return
+
+        if span.get_tag(SPAN_KIND) not in self._config.enabled_span_kinds:
+            return
+
+        if span.get_tag(self._config.tag_name):  # If the tag already exists, assume it is user generated
```

Comment:
This case (`peer.service` tag is already there so we set the source) should always be set regardless the span kind

Reply:
Swapping the order to make it so